raiden-ts/src/services/epics.ts

```diff
@@ -363,43 +364,53 @@ export const pfsServiceRegistryMonitorEpic = (
             undefined,
             confirmationBlocks,
             contractsInfo.ServiceRegistry.block_number,
-          ).pipe(
-            map(
-              // [service, valid_till, deposit_amount, deposit_contract, Event]
-              logToContractEvent<[Address, BigNumber, UInt<32>, Address, Event]>(
-                serviceRegistryContract,
+          )
+            .pipe(
+              withLatestFrom(latest$),
+              // filter only confirmed logs
+              filter(
+                ([{ blockNumber }, { state }]) =>
+                  !!blockNumber && blockNumber + confirmationBlocks <= state.blockNumber,
```

Comment:
I agree. The problem is that in each place, the `confirmationBlocks` and `blockNumber`/`txBlock` comes from different places, and are reactive values, so usually it may be hard to make a function which works for all cases, and as a simple expression, I've kept repeating myself, but let's consider this a nice refactor target for a future PR.